src/Rings/mpolyquo-localizations.jl

```diff
@@ -2698,27 +2698,54 @@ function Base.:(==)(
 
   if _has_coefficient_map(f)
     if _has_coefficient_map(g) 
-      coefficient_map(f) == coefficient_map(g) || return false
+      return is_equal_as_morphism(coefficient_map(f), coefficient_map(g))
     else
-      coefficient_map(f) == identity_map(coefficient_ring(domain(f))) || return false
+      return is_trivial(coefficient_map(f))
     end
   elseif _has_coefficient_map(g)
     if _has_coefficient_map(f) 
-      coefficient_map(f) == coefficient_map(g) || return false
+      return is_equal_as_morphism(coefficient_map(f), coefficient_map(g))
     else
-      coefficient_map(g) == identity_map(coefficient_ring(domain(g))) || return false
+      is_trivial(coefficient_map(g))
     end
   end
 
-  return all(x->f(x) == g(x), gens(domain(f)))
+  return all(f(x) == g(x) for x in gens(domain(f)))
+end
+
+# TODO: Move to wherever `MapFromFunc` comes from ?
+function is_equal_as_morphism(f::MapFromFunc, g::MapFromFunc)
+  f === g && return true
+  domain(f) === domain(g) || return false
+  codomain(f) === codomain(g) || return false
+  f.f === g.f && return true
+  error("comparison of maps $f and $g not possible")
+end
+
+function is_equal_as_morphism(f::Map, g::Map)
+  return f == g
+end
+
+function Base.:(==)(f::Map, g::Map)
+  f === g && return true
+  domain(f) === domain(g) || return false
+  codomain(f) === codomain(g) || return false
+  error("comparison of maps $f and $g not possible")
+end
```

Comment:
```suggestion
```
I think these are no longer needed here, as `is_equal_as_morphism` takes care of these cases already

Reply:
I found them to be necessary for some reason. But I will look into this again.